frame/composable-traits/src/oracle.rs

```diff
@@ -90,10 +93,17 @@ pub trait Oracle {
 		Self::get_price(asset, unit).map(|_| true)
 	}
 
+	/// Getting time weighted price for setted amount.
+	fn get_price_weighted(
+		asset_id: Self::AssetId,
+		amount: Self::Balance,
+	) -> Result<Self::Balance, DispatchError>;
+
 	/// Time Weighted Average Price
+	/// May be used to minimize a kurtosis impact on the price.
 	fn get_twap(
```

Comment:
I think we can just remove this function all together as there are no usages. Also specifying weights for prices externally seems like a bad idea